localization/ndt_scan_matcher/src/map_update_module.cpp

```diff
@@ -146,12 +146,25 @@ bool MapUpdateModule::update_ndt(const geometry_msgs::msg::Point & position, Ndt
     request,
     [](rclcpp::Client<autoware_map_msgs::srv::GetDifferentialPointCloudMap>::SharedFuture) {})};
 
+  // Wait for maximum 10 milliseconds
+  std::chrono::milliseconds timeout(10);
+  auto start = std::chrono::system_clock::now();
+
   std::future_status status = result.wait_for(std::chrono::seconds(0));
```

Comment:
@anhnv3991 
What does wait_for(std::chrono::seconds(0)) do?
Does it essentially do nothing by waiting for 0 seconds?
Or does it mean waiting indefinitely until completion?"

Reply:
@YamatoAndo No idea, it is not my code. Maybe the author wanted to check the status immediately after calling wait_for.

Reply:
So, how about writing the code like this?
```
while (result.wait_for(std::chrono::milliseconds(10)) != std::future_status::ready) {
  RCLCPP_WARN_STREAM_THROTTLE(
        logger_, *clock_, 1000, "Waited for incoming PCDs for too long...");
}
```